src/Stratis.Bitcoin.Features.Consensus/CoinViews/ICoinViewStorage.cs

```diff
@@ -0,0 +1,50 @@
+﻿using System.Collections.Generic;
+using System.Threading.Tasks;
+using NBitcoin;
+using Stratis.Bitcoin.Consensus;
+using Stratis.Bitcoin.Utilities;
+
+namespace Stratis.Bitcoin.Features.Consensus.CoinViews
+{
+    /// <summary>
+    /// Database of UTXOs.
+    /// </summary>
+    public interface ICoinViewStorage : ICoinView
+    {
+        /// <summary>
+        /// Persists changes to the coinview storage.
+        /// <para>
+        /// This method is provided (in <paramref name="unspentOutputs"/> parameter) with information about all
+        /// transactions that are either new or were changed in the new block. It is also provided with information
+        /// in original outputs about the previous state of those transactions (if any),
+        /// which is used for <see cref="ICoinView.Rewind"/> operation.
```

Comment:
It looks like you changed the logic, so this comment no longer reflects the reality